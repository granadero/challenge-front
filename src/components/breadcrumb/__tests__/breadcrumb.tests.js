import { render, screen } from "@testing-library/react";
import Breadcrumb from "../breadcrumb";

describe("Breadcrumb", () => {
  const categories = [
    { name: "Category 1" },
    { name: "Category 2" },
    { name: "Category 3" },
  ];

  it("renders the correct number of breadcrumb items", () => {
    render(<Breadcrumb categories={categories} />);
    const breadcrumbItems = screen.getAllByRole("listitem");
    expect(breadcrumbItems).toHaveLength(categories.length);
  });

  it("displays the separator between breadcrumb items", () => {
    render(<Breadcrumb categories={categories} />);
    const separator = screen.getAllByText(">");
    expect(separator[0]).toBeInTheDocument();
  });

  it("displays the correct breadcrumb names", () => {
    render(<Breadcrumb categories={categories} />);
    const breadcrumbNames = screen.getAllByText(/Category \d+/);
    expect(breadcrumbNames).toHaveLength(categories.length);
  });

  it("should match snapshot", () => {
    const { container } = render(<Breadcrumb categories={categories} />);
    expect(container).toMatchSnapshot();
  });
});
