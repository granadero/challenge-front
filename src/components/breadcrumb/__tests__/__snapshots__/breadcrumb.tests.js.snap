// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should match snapshot 1`] = `
<div>
  <section
    class="breadcrumbContainer"
  >
    <ul
      class="breadcrumb"
    >
      <li
        class="breadcrumbItem"
      >
        Category 1
        <span
          class="separator"
        >
          &gt;
        </span>
      </li>
      <li
        class="breadcrumbItem"
      >
        Category 2
        <span
          class="separator"
        >
          &gt;
        </span>
      </li>
      <li
        class="breadcrumbItem"
      >
        Category 3
      </li>
    </ul>
  </section>
</div>
`;
